Documentation generation with readthedocs.org

hansenms opened the pull request:
This PR is a first test at generating [readthedocs.org](https://readthedocs.org/) documentation for the ISMRMRD project. It relates the the documentation rework proposed in #188.

The main documentation has been moved from the README file to the docs folder and replaced with a reference.

kspaceKelvin commented on doc/requirements.txt:
```diff
@@ -0,0 +1,3 @@
+myst-parser
+sphinx~=4.3.0
+breathe
```
I think the following requirements would be helpful:
- ``sphinx-rtd-theme``: This would allow for local sphinx building during development
- `` sphinxcontrib-mermaid``: Used for mermaid diagrams in the docs

hansenms replied:
Should be added now.

hansenms commented:
@kspaceKelvin, I have updated the dependencies and I have also deleted the (now duplicated) information from the README.md. Could you give this a careful review?

hansenms commented:
@kspaceKelvin wanna get this merged?

kspaceKelvin reviewed (approved):
Looks good to me!  This should be a great starting point for documentation going forward.
